controller/src/beerocks/master/tasks/association_handling_task.cpp

```diff
@@ -70,18 +71,39 @@ void association_handling_task::work()
     case START_RSSI_MONITORING: {
         TASK_LOG(DEBUG) << "START_RSSI_MONITORING";
         /*
-            * request constant RSSI monitoring for the new client
-            */
+         * request constant RSSI monitoring for the new client
+         */
+
+        bool send_start_monitoring_request = true;
+        std::string new_hostap_mac;
+        std::string ipv4;
+
         if (database.is_node_wireless(sta_mac)) {
-            std::string new_hostap_mac = database.get_node_parent(sta_mac);
+            new_hostap_mac = database.get_node_parent(sta_mac);
             if (new_hostap_mac != original_parent_mac ||
                 database.get_node_state(sta_mac) != beerocks::STATE_CONNECTED) {
                 TASK_LOG(DEBUG) << "sta " << sta_mac << " is no longer connected to "
                                 << original_parent_mac << " finishing task";
                 finish();
                 return;
             }
-            std::string ipv4 = database.get_node_ipv4(sta_mac);
+
+            ipv4 = database.get_node_ipv4(sta_mac);
+
+            // Wait until ipv4 will be received on DHCP event or platform arp monitor notification
+            if (ipv4.empty()) {
+                if (std::chrono::steady_clock::now() >
+                    (task_started_timestamp + std::chrono::seconds(GET_IP_CLIENT_TIMEOUT_SEC))) {
+                    TASK_LOG(ERROR) << "ipv4 hasn't been received on client " << sta_mac
+                                    << ", skip start monitoring";
+                    send_start_monitoring_request = false;
+                } else {
+                    break;
```

Comment:
If you still decide to keep the current implementation (I hope you won't) - please at least add a comment here that explains the break is causing the state machine to stay in the same state.

Reply:
add a comment